libs/models/datapoints.py

```diff
@@ -0,0 +1,400 @@
+import datetime
+from abc import ABC, abstractmethod
+from logging import Logger
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, root_validator, validator
+
+from electricitymap.contrib.config import EXCHANGES_CONFIG, ZONES_CONFIG, ZoneKey
+from electricitymap.contrib.libs.models.constants import VALID_CURRENCIES
+
+LOWER_DATETIME_BOUND = datetime.datetime(2000, 1, 1, tzinfo=datetime.timezone.utc)
+
+
+class Mix(BaseModel, ABC):
+    @root_validator
+    def _validate_mix(cls, values: Dict[str, Optional[float]]):
+        if all(v is None for v in values.values()):
+            raise ValueError("Mix is completely empty")
+
+
+class ProductionMix(Mix):
+    biomass: Optional[float]
+    coal: Optional[float]
+    gas: Optional[float]
+    geothermal: Optional[float]
+    hydro: Optional[float]
+    nuclear: Optional[float]
+    oil: Optional[float]
+    solar: Optional[float]
+    unknown: Optional[float]
+    wind: Optional[float]
```

Comment:
Should these also accept int as a type or do we want all values to be floats?

Reply:
I will double check but I think if you supply an int, it just automatically gets cast to a float. Which is anyways an operation that we do afterwards because in the database they are stored as floats.